R/fimsframe.R

```diff
@@ -12,20 +12,82 @@
 
 setClass(
   Class = "FIMSFrame",
-  slots = c(data = "data.frame")
+  slots = c(
+    data = "data.frame", # can use c( ) or list here.
+    fleets = "numeric",
+    nyrs = "integer"
+  )
 )
 
-setClass("FIMSFrameAge",
-  slots = list(
-    weightatage = "data.frame",
-    ages = "numeric"
+# leaving FIMSFrameAge with just age related slots.
+setClass(
+  Class = "FIMSFrameAge",
+  slots = c(
+    ages = "numeric",
+    nages = "numeric",
```

Comment:
should this be "integer"?

Reply:
Probably, does integer work for a vector of integers? nages is a single value but ages is a vector.

Reply:
Yes, "integer" would work for a vector of integers
```
> is.vector(1:10)
[1] TRUE
> is.integer(1:10)
[1] TRUE
> is.numeric(1:10)
[1] TRUE
> is.numeric(c(1,1.1))
[1] TRUE
> is.integer(c(1,1.1))
[1] FALSE
```
but I am wondering if the vector has to be integers? Do we always want to assume that the vector of ages are integers as in 1, 2, 3 and not allow 1, 1.5, 2, 2.5, 3? Perhaps allowing half ages would be a feature implemented in the future and the model would break right now if a user tried to use half ages?

Reply:
I think we also assume ages are doubles in C++ code. See examples in [rcpp_growth.hpp](https://github.com/NOAA-FIMS/FIMS/blob/6b7200843d238080442b459b87c19855c5e9a9d9/inst/include/interface/rcpp/rcpp_objects/rcpp_growth.hpp#L43) and [rcpp_maturity.hpp](https://github.com/NOAA-FIMS/FIMS/blob/6b7200843d238080442b459b87c19855c5e9a9d9/inst/include/interface/rcpp/rcpp_objects/rcpp_maturity.hpp#L74).

Reply:
It makes sense then to leave ages as numeric.